tools/sigh.ts

```diff
@@ -723,6 +724,7 @@ function runTests(args: string[]): boolean {
     for (const test of tests) {
       chain.push(`
         import {mocha} from '${mochaInstanceFile}';
+        import {Flags} from '${flagsInstanceFile}';
```

Comment:
why is this needed? tests that need this can import the file?I would leave this change out unless you can provide a rationale for when to add imports here.

Reply:
Found out a bit more about this.
The problem is that Flags can be imported long before the `globals` test flags are set up, meaning that we don't incorporate the values into the `Flags` and get bad state (depending on test's interaction with `Flags`).

I've added a comment that we need the reset to update the flags with the command line flags.

Maybe there's a better way to do this (lazily getting the flags and command line flags at the first call) but I think it would be more complex and possibly add overhead to running and debugging any use of flags.

Is there a better way? Thanks for catching this, I hadn't thought it through enough when first using this as a fix.